src/transformation/utils/diagnostics.ts

```diff
@@ -86,6 +86,14 @@ export const invalidForRangeCall = createErrorDiagnosticFactory(
     (message: string) => `Invalid @forRange call: ${message}.`
 );
 
+export const invalidRangeUse = createErrorDiagnosticFactory(
+    "$range can only be used as a single expression in a for...of loop."
```

Comment:
Not sure what you mean with `as a single expression`, maybe just remove that part?